cpp/src/replace/replace.cu

```diff
@@ -87,6 +90,109 @@ __device__ auto get_new_value(cudf::size_type idx,
   return thrust::make_pair(new_value, output_is_valid);
 }
 
+__device__ int get_new_string_value(cudf::size_type idx,
+                                     cudf::column_device_view& input,
+                                     cudf::column_device_view& values_to_replace,
+                                     cudf::column_device_view& replacement_values) {
+  cudf::string_view input_string = input.element<cudf::string_view>(idx);
+  int match = -1;
+  for (int i = 0; i < values_to_replace.size(); i++) {
+    cudf::string_view value_string = values_to_replace.element<cudf::string_view>(i);
+    if (input_string == value_string){
+      match = i;
+      break;
+    }
+  }
+  return match;
+}
+
+template<bool input_has_nulls, bool replacement_has_nulls>
+__global__ void replace_strings_first_pass(cudf::column_device_view input,
+                                           cudf::column_device_view values_to_replace,
+                                           cudf::column_device_view replacement,
+                                           cudf::mutable_column_device_view offsets,
+                                           cudf::mutable_column_device_view indices) {
+  cudf::size_type nrows = input.size();
+  cudf::size_type i = blockIdx.x * blockDim.x + threadIdx.x;
+  while (i < nrows) {
+    bool input_is_valid = true;
+    if (input_has_nulls)
+      input_is_valid = input.is_valid_nocheck(i);
+    if (input_is_valid){
```

Comment:
Suggestion, delete lines 118-120 and
```suggestion
    if (input_has_nulls && input.is_valid_nocheck(i)){
```

Reply:
That doesn't take care of the case where input_has_nulls is false, I don't think.

Reply:
You are right. This would, I think.

```suggestion
    if (!input_has_nulls || input.is_valid_nocheck(i)){
```